oracle/tests/test_oracle.py

```diff
@@ -1,288 +1,310 @@
-
-
-def test__get_tablespace_metrics(aggregator, check):
-    query = "SELECT TABLESPACE_NAME, sum(BYTES), sum(MAXBYTES) FROM sys.dba_data_files GROUP BY TABLESPACE_NAME"
-
-    con = mock.MagicMock()
-    cur = mock.MagicMock()
-    cur.fetchall.return_value = [
-        ["offline", None, 100],
-        ["normal", 50, 100],
-        ["full", 100, 100],
-        ["size_0", 1, None]
-    ]
-    con.cursor.return_value = cur
-
-    check._get_tablespace_metrics(con, ["custom_tag"])
-    cur.execute.assert_called_with(query)
-
-    # Offline tablespace
-    tags = ["custom_tag", "tablespace:offline"]
-    aggregator.assert_metric("oracle.tablespace.used", value=0, count=1, tags=tags)
-    aggregator.assert_metric("oracle.tablespace.size", value=100, count=1, tags=tags)
-    aggregator.assert_metric("oracle.tablespace.in_use", value=0, count=1, tags=tags)
-    aggregator.assert_metric("oracle.tablespace.offline", value=1, count=1, tags=tags)
-
-    # Normal tablespace
-    tags = ["custom_tag", "tablespace:normal"]
-    aggregator.assert_metric("oracle.tablespace.used", value=50, count=1, tags=tags)
-    aggregator.assert_metric("oracle.tablespace.size", value=100, count=1, tags=tags)
-    aggregator.assert_metric("oracle.tablespace.in_use", value=50, count=1, tags=tags)
-    aggregator.assert_metric("oracle.tablespace.offline", value=0, count=1, tags=tags)
-
-    # Full tablespace
-    tags = ["custom_tag", "tablespace:full"]
-    aggregator.assert_metric("oracle.tablespace.used", value=100, count=1, tags=tags)
-    aggregator.assert_metric("oracle.tablespace.size", value=100, count=1, tags=tags)
-    aggregator.assert_metric("oracle.tablespace.in_use", value=100, count=1, tags=tags)
-    aggregator.assert_metric("oracle.tablespace.offline", value=0, count=1, tags=tags)
-
-    # Size 0 tablespace
-    tags = ["custom_tag", "tablespace:size_0"]
-    aggregator.assert_metric("oracle.tablespace.used", value=1, count=1, tags=tags)
-    aggregator.assert_metric("oracle.tablespace.size", value=0, count=1, tags=tags)
-    aggregator.assert_metric("oracle.tablespace.in_use", value=100, count=1, tags=tags)
-    aggregator.assert_metric("oracle.tablespace.offline", value=0, count=1, tags=tags)
+class TestConfig:
+    def test__get_config(self, check, instance):
+        """
+        Test the _get_config method
+        """
+        server, user, password, service, jdbc_driver, tags, custom_queries = check._get_config(instance)
+        assert user == 'system'
+        assert password == 'oracle'
+        assert service == 'xe'
+        assert jdbc_driver is None
+        assert tags == ['optional:tag1']
+        assert custom_queries == []
+        assert check.server == 'localhost:1521'
+
+    def test_check_misconfig(self, check, instance):
+        """
+        Test bad config values
+        """
+        instance['server'] = None
+        with pytest.raises(OracleConfigError):
+            check.check(instance)
+
+
+class TestClientFallback:
```

Comment:
why did you switch to test classes?

Reply:
Good question!

I think it's easier to organize the tests when you can group them and I also think the output of the tests looks a lot cleaner.

Reply:
You can split the test functions across multiple modules, logically grouped. This is what we do in all the other integrations, I'd like to keep a consistent style.

Reply:
You got it, I'll update the tests 🏎